### PR TITLE
Rename mestastable states

### LINKED ISSUE #392: rename `metastable_states` to `macrostates`
- [ ] Additional function parameters, change functionality or change defaults?
- [ ] New estimator in `cr.tl.estimators`?
- [ ] New kernel in `cr.tl.kernels`?
- [ ] New gene trend model in `cr.ul.models`?
- [ ] New plotting function in `cr.pl`?
- [x] Other?

We have finally reached an agreement here. `macrostates` is the term we would like to use. Is that okay for you @michalk8 ?

### PR DESCRIPTION
**Pull request by michalk8**
**IMPORTANT: Please search among the [Pull request](../pulls) before creating one.**

## Title

Rename `metastable states` -> `macrostates`.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] New feature (non-breaking change which adds functionality)

## Closes

closes #392 

## TODOs
- [ ] @Marius1311 could you please add some text for GPCCA estimator?
- [ ] make release notes
- [ ] update tutorials (after merging this PR)

After merging, I will make a new release.

### CODE REVIEW

**Comment by Marius1311** [comment]
Sure! Just point me to the specific function/method please.

**Comment by michalk8** [comment]
It's for the init method of GPCCA: https://github.com/theislab/cellrank/blob/master/cellrank/tl/estimators/_gppca.py#L37

**Comment by michalk8** [comment]
Hi @Marius1311 , when you find the chance, please included the text for the GPCCA.
I will in the meantime write some tests for #395 .

**Comment by Marius1311** [comment]
Done! Added some description for GPCCA.

File: cellrank/tl/_init_term_states.py [hunk 1]
```diff
@@ -27,11 +27,11 @@
 on the velocity vectors of each individual cell. Based on this Markov chain, we provide two estimators
 to compute {direction} states, both of which are based on spectral methods.
 
-For the estimator :class:`cellrank.tl.estimators.GPCCA`, cells are fuzzily clustered into metastable states,
+For the estimator :class:`cellrank.tl.estimators.GPCCA`, cells are fuzzily clustered into macrostates,
 using Generalized Perron Cluster Cluster Analysis [GPCCA18]_.
 In short, this coarse-grains the Markov chain into a set of macrostates representing the slow
 time-scale dynamics, i.e. transitions between these macrostates are rare. The most stable ones of these will represent
-{direction}, while the others will represent transient, metastable states.
+{direction}, while the others will represent transient, macrostates.
```
Comment (Marius1311) [inline cellrank/tl/_init_term_states.py:34]: better: "while the others represent intermediate macrostates"

**Review by Marius1311 — APPROVED** [review]
This looks fine!

**Comment by michalk8** [comment]
Previous checks have passed, merging this.
